smarts/core/road_map.py

```diff
@@ -290,14 +290,18 @@ def center_at_point(self, point: Point) -> Point:
         def edges_at_point(self, point: Point) -> Tuple[Point, Point]:
             offset = self.offset_along_lane(point)
             width = self.width_at_offset(offset)
-            left_edge = RefLanePoint(s=offset, t=width / 2)
-            right_edge = RefLanePoint(s=offset, t=-width / 2)
-            return (self.from_lane_coord(left_edge), self.from_lane_coord(right_edge))
+            left_edge = RefLinePoint(s=offset, t=width / 2)
+            right_edge = RefLinePoint(s=offset, t=-width / 2)
+            return self.from_lane_coord(left_edge), self.from_lane_coord(right_edge)
 
         def vector_at_offset(self, start_offset: float) -> np.ndarray:
-            add_offset = 1  # a little further down the lane
-            end_offset = start_offset + add_offset
-            p1 = self.from_lane_coord(RefLinePoint(s=start_offset))
+            if start_offset >= self.length:
+                s_offset = self.length - 1
```

Comment:
This is still problematic if `self.length` is less than 1.  Probably best would be to replace the entire `if`/`else` block with:
```
before_offset = max(start_offset - 0.5, 0)
after_offset = min(start_offset + 0.5, self.length)
if before_offset == after_offset:
    return np.array((0,0,0))
p1 = self.from_lane_coord(RefLinePoint(s=before_offset))
p2 = self.from_lane_coord(RefLinePoint(s=after_offset))
return np.array(p2) - np.array(p1)
```

(Note that for most map formats, this "compare to the same point offset by a little" technique is un-necessary because the vector is given by the direction of the nearest segment.)

Reply:
... and because this can still return (0,0,0) in rare occasions (for example, where a lane has 0 length), we should still also add a check downstream to prevent the divide-by-zero exception.

Reply:
On 

> This is still problematic if `self.length` is less than 1. Probably best would be to replace the entire `if`/`else` block with:

On second thought, given that we don't want to change the behavior for existing code/scenarios, I guess we should only do the above when `start_offset` is greater than or equal to `self.length`.  

So I withdraw my suggested code (but you will still need to add a `max` to prevent `s_offset` from being less than 0).

Reply:
ok, sure!